core/unix/signal.c

```diff
@@ -537,7 +537,16 @@ signal_thread_init(dcontext_t *dcontext, void *os_data)
         signal_frame_extra_size(true)
         /* sigpending_t has xstate inside it already */
         IF_LINUX(IF_X86(-sizeof(kernel_xstate_t)));
-    IF_LINUX(IF_X86(ASSERT(!YMM_ENABLED() || ALIGNED(pend_unit_size, AVX_ALIGNMENT))));
+/* Above does not guarantee alignment, it only guarantees sufficient space including
+ * aligmment of the extended state. The following is needed in order to guarantee
+ * alignment of our pending unit's size, which is used as the block size in the
+ * special allocator.
+ */
+#ifdef X86
+    pend_unit_size =
+        ALIGN_FORWARD(pend_unit_size, YMM_ENABLED() ? AVX_ALIGNMENT : FPSTATE_ALIGNMENT);
```

Comment:
It seems like it's too big now: can we remove signal_frame_extra_size from line 537?  That should only be needed when placing xstate separately.  It seems like it isn't needed at all for pending?  Also, if we have special heap align forward for us, we don't need this align here either.

Reply:
I actually think it's fine now. sizeof(sigpending_t) - sizeof(kernel_xstate_t) provides storage for all the sigpending stuff minus xstate. The signal_frame_extra_size(true) adds enough storage to be able to run xsave, based on the real measured size. In any case if we touch it, maybe in a different PR since this is unrelated? Re-open if necessary.